Ginger/GingerCore/Actions/ActLaunchJavaWSApplication.cs

```diff
@@ -325,10 +325,28 @@ public override void Execute()
 
             if (mLaunchWithAgent)
             {
-                //For Parallel execution, we want attach to be synchronized.
-                //So for windows with same title, correct process id will be calculated
-                lock (syncLock)
+                try
                 {
+                    //For Parallel execution (also between diffrent Ginger processes), we want attach to be synchronized.
+                    //So for windows with same title, correct process id will be calculated
+                    try
+                    {
+                        // acquire the mutex (or timeout), will return false if it timed out
+                        Reporter.ToLog(eLogLevel.DEBUG, "Attach Java Agent- Waiting for Mutex Release");
+                        if (!mutex.WaitOne(120000))//TODO: set timeout as mWaitForWindowTitleMaxTime_Calc?
+                        {
+                            Reporter.ToLog(eLogLevel.WARN, "Attach Java Agent- Mutex Wait Timeout Reached");
+                        }
+                        else
+                        {
+                            Reporter.ToLog(eLogLevel.DEBUG, "Attach Java Agent- Mutex was Released");
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Reporter.ToLog(eLogLevel.WARN, "Attach Java Agent- Mutex Wait Threw Exception", ex); ;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5145905)